tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/SdkRepoConfigHelper.cs

```diff
@@ -0,0 +1,210 @@
+                using var configJson = JsonDocument.Parse(configContent);
+
+                // Use helper method to navigate JSON path
+                var (found, element) = TryGetJsonElementByPath(configJson.RootElement, jsonPath);
+                if (!found)
+                {
+                    throw new InvalidOperationException($"Property not found at JSON path '{jsonPath}' in configuration file {specToSdkConfigFilePath}.");
+                }
+
+                // Deserialize to the requested type
+                var value = JsonSerializer.Deserialize<T>(element.GetRawText());
+                if (value == null)
+                {
+                    throw new InvalidOperationException($"Failed to deserialize value at JSON path '{jsonPath}' to type {typeof(T).Name}.");
+                }
+
+                _logger.LogDebug("Retrieved config value from {JsonPath} as {Type}", jsonPath, typeof(T).Name);
+                return value;
+            }
+            catch (JsonException ex)
+            {
+                throw new InvalidOperationException($"Error parsing JSON configuration: {ex.Message}", ex);
+            }
+        }
+
+        // Get build configuration (either command or script path)
+        public async Task<(BuildConfigType type, string value)> GetBuildConfigurationAsync(string repositoryRoot)
+        {
+            // Try command first
+            try
+            {
+                var command = await GetConfigValueFromRepoAsync<string>(repositoryRoot, BuildCommandJsonPath);
+                if (!string.IsNullOrEmpty(command))
+                {
+                    _logger.LogDebug("Found build command configuration");
+                    return (BuildConfigType.Command, command);
+                }
+            }
+            catch (InvalidOperationException)
+            {
+                // Command not found, continue to try path
+                _logger.LogDebug("No build command found, trying script path");
+            }
+
+            // Try path
+            try
+            {
+                var path = await GetConfigValueFromRepoAsync<string>(repositoryRoot, BuildScriptPathJsonPath);
+                if (!string.IsNullOrEmpty(path))
+                {
+                    _logger.LogDebug("Found build script path configuration");
+                    return (BuildConfigType.ScriptPath, path);
+                }
+            }
+            catch (InvalidOperationException)
+            {
+                // Path not found either
+                _logger.LogError("No build configuration found");
+            }
+
+            throw new InvalidOperationException($"Neither '{BuildCommandJsonPath}' nor '{BuildScriptPathJsonPath}' found in configuration.");
+        }
+
+        // Substitute template variables in command strings
+        public string SubstituteCommandVariables(string command, Dictionary<string, string> variables)
```

Comment:
Are these command strings intended to come from a future config file?

Reply:
Yes, the command could be arbitrary across languages.